### PR TITLE
cdc: flush redo meta in owner instead of processor

### LINKED ISSUE #6277: cdc redo meta can be broken if cdc fails when executing DDLs
### Before asking a question, make sure you have

- [X] Searched existing Stack Overflow questions
- [X] Googled your question
- [X] Searched open and closed [GitHub issues](https://github.com/pingcap/tiflow/issues?q=is%3Aissue)
- [X] Read the [documentation](https://docs.pingcap.com/tidb/stable)

### What is your question?

This bug is introduced with https://github.com/pingcap/tiflow/pull/6207.

Here is a time line example which can meet the bug:
1. for a given cdc instance replicating table A, `A.resolvedTimestamp` = 10;
2. a DDL with timestamp 11 is in executing;
3. no more row events with timestamp higher than 11 can be flushed, but resolved timestamp of table A can still be advanced to 20;
4. shutdown the cdc instance, and run `cdc redo apply` for it;
5. `cdc redo apply` will get `resolved timestamp = 20` from the redo meta, which is wrong.

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Issue Number: close #6277 

### What is changed and how it works?

Flush redo meta in owner instead of processor.

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/redo/manager.go [hunk 1]
```diff
@@ -169,6 +170,11 @@ type ManagerImpl struct {
 	// and flushed is updated in flushLog.
 	rtsMap sync.Map
 
+	gotCheckpointTs     model.Ts
```
Comment (CharlesCheung96) [inline cdc/redo/manager.go:173]: Could `model.Ts` be replaced with `statefulRts` here?

File: cdc/redo/manager.go [hunk 2]
```diff
@@ -354,10 +358,16 @@ func (m *ManagerImpl) GetMinResolvedTs() model.Ts {
 	return atomic.LoadUint64(&m.minResolvedTs)
 }
 
-// UpdateCheckpointTs updates checkpoint-ts to redo log writer
-func (m *ManagerImpl) UpdateCheckpointTs(ctx context.Context, checkpointTs uint64) (err error) {
-	err = m.writer.EmitCheckpointTs(ctx, checkpointTs)
-	return
+// UpdateMeta updates meta.
+func (m *ManagerImpl) UpdateMeta(checkpointTs, resolvedTs model.Ts) {
+	atomic.StoreUint64(&m.metaResolvedTs.unflushed, resolvedTs)
```
Comment (CharlesCheung96) [inline cdc/redo/manager.go:363]:
```suggestion
	m.metaResolvedTs.checkAndSetUnflushed(resolvedTs)
	m.metaCheckpointTs.checkAndSetUnflushed(checkpointTs)
```

File: cdc/redo/writer/writer.go [hunk 3]
```diff
@@ -579,8 +537,8 @@ func (l *LogWriter) mergeMeta(checkPointTs uint64, rtsMap map[model.TableID]mode
 	return data, err
 }
 
-func (l *LogWriter) flushLogMeta(checkPointTs uint64, rtsMap map[model.TableID]model.Ts) error {
-	data, err := l.mergeMeta(checkPointTs, rtsMap)
+func (l *LogWriter) flushLogMeta(checkpointTs, resolvedTs uint64) error {
+	data, err := l.mergeMeta(checkpointTs, resolvedTs)
```
Comment (CharlesCheung96) [inline cdc/redo/writer/writer.go:541]: The `mergeMeta` name is somewhat confusing.

File: cdc/redo/writer/writer.go [hunk 4]
```diff
@@ -579,8 +537,8 @@ func (l *LogWriter) mergeMeta(checkPointTs uint64, rtsMap map[model.TableID]mode
 	return data, err
 }
 
-func (l *LogWriter) flushLogMeta(checkPointTs uint64, rtsMap map[model.TableID]model.Ts) error {
-	data, err := l.mergeMeta(checkPointTs, rtsMap)
+func (l *LogWriter) flushLogMeta(checkpointTs, resolvedTs uint64) error {
+	data, err := l.maybeUpdateMeta(checkpointTs, resolvedTs)
```
Comment (nongfushanquan) [inline cdc/redo/writer/writer.go:541]: if no change happened , should it return here ?

File: cdc/owner/changefeed.go [hunk 5]
```diff
@@ -432,7 +452,7 @@ LOOP:
 	}()
 
 	stdCtx := contextutil.PutChangefeedIDInCtx(cancelCtx, c.id)
-	redoManagerOpts := &redo.ManagerOptions{EnableBgRunner: false}
+	redoManagerOpts := &redo.ManagerOptions{EnableBgRunner: true}
```
Comment (CharlesCheung96) [inline cdc/owner/changefeed.go:455]: Ref: https://github.com/pingcap/tiflow/blob/master/cdc/owner/changefeed.go#L527

Comment (hicqu) [reply cdc/owner/changefeed.go:455]: At that place, the redoManager is created for clean something after the changefeed has been removed. So `false` is OK.

**Review by CharlesCheung96 — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

File: cdc/redo/common/redo.go [hunk 6]
```diff
@@ -59,59 +56,20 @@ const (
 
 // LogMeta is used for store meta info.
 type LogMeta struct {
-	CheckPointTs   uint64
-	ResolvedTsList map[int64]uint64
+	CheckpointTs uint64 `msg:"checkPointTs"`
```
Comment (zhaoxinyu) [inline cdc/redo/common/redo.go:59]:
```suggestion
	CheckpointTs uint64 `msg:"checkpointTs"`
```

**Pull request merged**